cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -325,19 +345,39 @@ private HttpRequest createGetRefreshTokenRequest(OAuthProvider provider, String
   private HttpRequest createGetAccessTokenRequest(OAuthProvider provider, String refreshToken)
       throws OAuthServiceException {
     OAuthClientCredentials clientCreds = provider.getClientCredentials();
-    try {
-      return HttpRequest.post(new URL(provider.getTokenRefreshURL()))
-          .withBody(
-              String.format("grant_type=refresh_token&client_id=%s&client_secret=%s&refresh_token=%s",
+    CredentialEncodingStrategy strategy = provider.getCredentialEncodingStrategy();
+        
+    String body;
+    switch (strategy) {
+      case BASIC_AUTH:
+        body = String.format("grant_type=refresh_token&refresh_token=%s", refreshToken);
+      case FORM_BODY:
+      default:
+        body = String.format("grant_type=refresh_token&client_id=%s&client_secret=%s&refresh_token=%s",
                 clientCreds.getClientId(),
                 clientCreds.getClientSecret(),
-                refreshToken))
+                refreshToken);
+    }
+
+    try {
+      HttpRequest.Builder request = HttpRequest.post(new URL(provider.getTokenRefreshURL()))
+          .withBody(body)
           .build();
```

Comment:
This looks like a compilation issue, if we are assigning it to `HttpRequest.Builder`, we cannot do `build` here.